docs/pages/deploy-a-cluster/high-availability.mdx

```diff
@@ -0,0 +1,369 @@
+  Encrypt.
+
+{/* TODO: include a diagram--do this after filling in the rest of the details*/}
+
+## Layer Four load balancer
+
+The load balancer forwards traffic from users and services to an available
+Teleport instance. This must not terminate TLS, and must transparently forward
+the TCP traffic it receives. In other words, this must be a Layer Four load
+balancer, not a Layer Seven (e.g., HTTP) load balancer. 
+
+Configure the load balancer to forward traffic from the following ports on the
+load balancer to the corresponding port on an available Teleport instance:
+
+| Port | Description |
+| - | - |
+| `3021` | Port used by Teleport Proxy Service instances to dial agents in [Proxy Peering mode](../management/operations/proxy-peering.mdx). |
+| `3022` | SSH port. This is Teleport's equivalent of port `22` for SSH. |
+| `3023` | SSH port for clients connect to. The Proxy Service will forward this connection to port `3022` on the destination Node. |
+| `3024` | SSH port used to create reverse SSH tunnels from behind-firewall environments. |
+| `3025` | TLS port used by the Auth Service to serve its gRPC API to Teleport services  in a cluster. |
+| `443` | HTTPS connections to authenticate `tsh` users into the cluster. The same connection is used to serve a Web UI. |
+| `3026` | HTTPS Kubernetes proxy |
+| `3027` | Kubernetes Service |
+| `3028` | Desktop Service |
+| `3036` | MySQL port |
+| `5432` | Postgres port |
+
+## Key-value store
+
+The Teleport Auth Service stores cluster state (such as dynamic configuration
+resources) and audit events as key/value pairs. In high-availability
+deployments, you must configure the Auth Service to manage this data in a
+key-value store that runs outside of your cluster of Teleport instances.
+
+The Auth Service supports the following backends for cluster state and audit
+events:
+
+- Amazon DynamoDB
+- Google Cloud Firestore
+- etcd
+
+For Amazon DynamoDB and Google Cloud Firestore, your Teleport configuration (which
+we will describe in more detail in the [Configuration](#configuration) section)
+names a table or collection where Teleport stores cluster state and audit
+events. For etcd, Teleport uses namespaces within item keys to identify audit
+events and cluster state data.
+
+The Teleport Auth Service manages the creation of any required DynamoDB tables
+or Firestore collections itself, and does not require them to exist in advance.
+
+<Admonition title="Required permissions">
+
+In your cloud provider's RBAC solution (e.g., AWS or Google Cloud IAM), your
+Teleport instances need permissions to read from and write to your chosen
+key/value store, as well as to create tables and collections (if your key/value
+store supports them).
+
+</Admonition>
+
+## Object storage service
+
+High-availability Teleport deployments use an object storage service for
+persisting session recordings. The Teleport Auth Service supports two object
+storage services:
```

Comment:
We support any S3 compatible object storage.  On-prem customers often opt for MinIO or Ceph as solutions outside of the major cloud providers